lib/timers.js

```diff
@@ -481,6 +481,7 @@ Timeout.prototype.unref = function() {
     }
 
     var handle = reuse(this);
+    delete handle._list;
```

Comment:
> > Why not removing the reference to the TimersList instance in reuse's implementation instead?

> Messes with some tests that check things after unenroll(), which also uses reuse().

Making that change (clearing the handle's reference to the `TimersList` instance) makes only one test fail on `master`: `test/parallel/test-timers-same-timeout-wrong-list-deleted.js`. I think that test could be rewritten to not rely on that reference being present for unenrolled timers.

Is this what you had in mind by "Messes with some tests"?